src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -178,8 +178,12 @@ private void ParseDocumentAsync(Document document)
 
             var cancellationToken = cancellationTokenSource.Token;
 
+            // once task runs, we never cancel running parsing request. this will make sure 
+            // chained incremental parsing request will always get shorten by BG
+            //
+            // we are alraedy running in BG, you no need to run async version
             var task = _taskScheduler.ScheduleTask(
-                () => document.GetSyntaxTreeAsync(cancellationToken),
+                () => document.GetSyntaxTreeAsync(CancellationToken.None),
                 "BackgroundParser.ParseDocumentAsync",
                 cancellationToken);
```

Comment:
It would probably be good to comment why we *do* want a cancellation token on this line.  That will help future readers understand that it's ok for these two to be different.

Reply:
Agreed, it's not clear if this is intentional or a bug. I can imagine the reason is "once we've started, we won't stop, but if we never started it's OK to cancel work".